### ISSUE TITLE
Upgrade to 3.0.1 - Failed on 2 Sites

### ISSUE THREAD

**Issue by Rickcarlson103** [formtools/core#211]
Hi.  You helped me get my upgrade files to download earlier today.  I am upgrading from Formtools 2.2.6 to 3.0.1.  I logged out of Formtools, unzipped the the files and carefully uploaded the new files (a confirmed total of 2236 files) into my website formtools directory using Filezilla.  As expected, they were over-writing the old files.  I then tried to refresh the login page to log into formtools but it just says "This page isn’t working rickcarlson.com is currently unable to handle this request.
HTTP ERROR 500"  The url path is http://www.rickcarlson.com/formtools.   I have a 2nd website that I was trying to upgrade as will and thought something must have gone wrong with the first website so I tried it on the other.  I get the exact same problem with it as well.  http://www.722club.com/formtools. 
Fortunately, I ran full website backups before I did this but I'm wondering if you have any ideas as to what went wrong.  Thanks.

**Comment by Rickcarlson103** [comment]
It's an ordinary installation.  No customizing in case that helps.

**Comment by benkeen** [comment]
Hey @Rickcarlson103, sorry you ran into trouble! 

Couple of things:
- what version of PHP are you running? Form Tools 3.x supports PHP 5.3 and up. From what you described, I wonder if maybe you're running an older version? 
- If the previous issue isn't the cause, could you check your server logs to see what error is being logged? This'll be extremely helpful - it should help pinpoint it pretty quickly. 

Ben

**Comment by Rickcarlson103** [comment]
Thanks.  I will find out from the host company in the morning what version of php is running.  It’s possible it is something older because before I transitioned to formtools I used another application instead and it wasn’t being updated anymore (formfields) and I had to ask the host to keep an older version of php running.  Thanks. 

 

Richard Carlson | Cuelenaere LLP | Suite 200 Nexus Building, 450 2nd Ave North, Saskatoon SK  S7K 2C3 | t 306.477.7259 | f 306.652.4171 | www.rickcarlson.com | www. cuelenaere.com | mail@rickcarlson.com

**Comment by Rickcarlson103** [comment]
Hi.  I can see it is running php 5.3.  I have attached a screen shot showing php settings.  Maybe that’s of no use to you but I thought I would send it in case it reveals something to you?  If not, then I will check the logs next.  Thanks for the help.

 

Richard Carlson, Cuelenaere LLP | Suite 200 Nexus Building, 450 – 2nd Ave North, Saskatoon SK  S7K 2C3 | mail@rickcarlson.com |  www.rickcarlson.com | www.cuelenaere.com

**Comment by Rickcarlson103** [comment]
Does this reveal anything?  I’m not experienced at going through error logs but could only find this:

 

[11-May-2018 14:16:29 America/Chicago] PHP Warning:  session_start() [<a href='function.session-start'>function.session-start</a>]: Cannot send session cookie - headers already sent by (output started at /home/rcarlson/public_html/linkscourt_articles.inc:16) in /home/rcarlson/public_html/formtools/modules/form_builder/global/code/generator.php on line 681

[11-May-2018 14:16:29 America/Chicago] PHP Warning:  session_start() [<a href='function.session-start'>function.session-start</a>]: Cannot send session cache limiter - headers already sent (output started at /home/rcarlson/public_html/linkscourt_articles.inc:16) in /home/rcarlson/public_html/formtools/modules/form_builder/global/code/generator.php on line 681

[11-May-2018 14:16:29 America/Chicago] PHP Warning:  Cannot modify header information - headers already sent by (output started at /home/rcarlson/public_html/linkscourt_articles.inc:16) in /home/rcarlson/public_html/formtools/modules/form_builder/global/code/generator.php on line 682

[11-May-2018 14:16:29 America/Chicago] PHP Warning:  Cannot modify header information - headers already sent by (output started at /home/rcarlson/public_html/linkscourt_articles.inc:16) in /home/rcarlson/public_html/formtools/modules/form_builder/global/code/generator.php on line 683

 

Richard Carlson, Cuelenaere LLP | Suite 200 Nexus Building, 450 – 2nd Ave North, Saskatoon SK  S7K 2C3 | mail@rickcarlson.com |  www.rickcarlson.com | www.cuelenaere.com

**Comment by benkeen** [comment]
Yeah that helps, thanks! All those problem lines stem from an old version of the Form Builder: it shouldn't be loading that generator.php file at all - it was removed in Form Tools 3. 

Could you check your download package you got from the site? Unzip it and go to `/modules/form_builder`. Does that folder exist? 

It looks like either the download package didn't include the latest files for that module, or if it did, when you copied them to your server they didn't survive the trip. Curiouser and curiouser...

**Comment by Rickcarlson103** [comment]
Yes, formtools/modules/form_builder exists.  It’s in the unzipped package.  It was generated for both of the 2 websites when I clicked upgrade and downloaded the recommended zip file it created.

 

Richard Carlson | Cuelenaere LLP | Suite 200 Nexus Building, 450 2nd Ave North, Saskatoon SK  S7K 2C3 | t 306.477.7259 | f 306.652.4171 | www.rickcarlson.com | www. cuelenaere.com | mail@rickcarlson.com

**Comment by Rickcarlson103** [comment]
![image](https://user-images.githubusercontent.com/39168507/40035980-f1c47cb0-57c1-11e8-93c5-790a8662975a.png)
This is a screenshot after formtools/modules    
You'll see that form_builder exists

Then here are the contents of the form_builder directory in both cases:
![image](https://user-images.githubusercontent.com/39168507/40036018-334403ea-57c2-11e8-9473-f199840f8a78.png)

**Comment by Rickcarlson103** [comment]
I went into one of the 2 sites that had failed.  Before restoring to v 2.6** I renamed the v3.0.1 directory so it would remain.  Expanding it, here are the contents of the form_builder directory that uploaded
![image](https://user-images.githubusercontent.com/39168507/40036151-e89c59b8-57c2-11e8-9347-3a8888e2970d.png)
